src/OrchardCore/OrchardCore/Localization/NullLocalizerFactory.cs

```diff
@@ -29,18 +35,20 @@ private class NullLocalizer : IStringLocalizer, IHtmlLocalizer
             {
                 get
                 {
-                    var value = String.Empty;
+                    var translation = name;
+
                     if (arguments.Length == 1 && arguments[0] is PluralizationArgument pluralArgument)
                     {
-                        var pluralForm = pluralArgument.Forms[_defaultPluralRule(pluralArgument.Count)];
-                        value = String.Format(pluralForm, pluralArgument.Count);
-                    }
-                    else
-                    {
-                        value = String.Format(name, arguments);
+                        translation = pluralArgument.Forms[_defaultPluralRule(pluralArgument.Count)];
+
+                        arguments = new object[pluralArgument.Arguments.Length + 1];
+                        arguments[0] = pluralArgument.Count;
+                        Array.Copy(pluralArgument.Arguments, 0, arguments, 1, pluralArgument.Arguments.Length);
                     }
 
-                    return new LocalizedString(name, value);
+                    translation = String.Format(translation, arguments);
```

Comment:
This always will overwrite the first assignment

Reply:
Yes, but look again, the `translation` is what is passed as an argument too.